test/dynamo/test_graph_break_messages.py

```diff
@@ -0,0 +1,503 @@
+
+    def test_dynamo_graph_break_fn_with_msg(self):
+        def fn():
+            torch._dynamo.graph_break(msg="test graph break")
+
+        self.assertExpectedInlineMunged(
+            Unsupported,
+            lambda: torch.compile(fn, backend="eager", fullgraph=True)(),
+            """\
+Call to `torch._dynamo.graph_break()`
+  Explanation: User-inserted graph break. Message: test graph break
+  Hint: Remove the `torch._dynamo.graph_break()` call.
+
+  Developer debug context: Called `torch._dynamo.graph_break()` with args `[]`, kwargs `{'msg': ConstantVariable(str: 'test graph break')}`
+
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    torch._dynamo.graph_break(msg="test graph break")""",
+        )
+
+    def test_warnings(self):
+        def fn():
+            warnings.warn("test")
+
+        self.assertExpectedInlineMunged(
+            Unsupported,
+            lambda: torch.compile(fn, backend="eager", fullgraph=True)(),
+            """\
+Attempted to call function marked as skipped
+  Explanation: Dynamo does not know how to trace the Python builtin `_warnings.warn`.
+  Hint: If you are attempting to call a logging function (e.g. `_warnings.warn`), you can try adding it to `torch._dynamo.config.reorderable_logging_functions`.
+  Hint: Please file an issue on GitHub so the PyTorch team can add support for it.
+
+  Developer debug context: module: _warnings, qualname: warn, skip reason: <missing reason>
+
+
+from user code:
+   File "test_graph_break_messages.py", line N, in fn
+    warnings.warn("test")""",
+        )
+
+    @unittest.skipIf(not python_pytree._cxx_pytree_exists, "missing optree package")
+    def test_optree_graph_break_message(self):
+        import optree
+
+        @torch.compile(backend="eager")
+        def fn(x):
+            d = {"a": 1}
+            optree.tree_flatten(d)
+            return torch.sin(x)
+
+        fn(torch.randn(4))
+        self.assertEqual(len(counters["graph_break"]), 1)
+        first_graph_break = next(iter(counters["graph_break"].keys()))
+        self.assertExpectedInline(
+            first_graph_break,
+            """\
+Attempted to call function marked as skipped
+  Explanation: Dynamo cannot trace optree C/C++ function optree._C.PyCapsule.flatten.
+  Hint:  Consider using torch.utils._pytree - https://github.com/pytorch/pytorch/blob/main/torch/utils/_pytree.py
+
+  Developer debug context: module: optree._C, qualname: PyCapsule.flatten, skip reason: <missing reason>
+""",
+        )
+
+    @scoped_load_inline
+    @torch._dynamo.config.patch(inline_inbuilt_nn_modules=False)
+    @unittest.skipIf(IS_FBCODE, "inline cpp_extension doesn't work in fbcode")
+    def test_cpp_extension_recommends_custom_ops(self, load_inline):
+        cpp_source = """
+        #include <torch/extension.h>
+        at::Tensor foobar(const at::Tensor& x) {
+            return x.clone();
+        }
+        """
+        module = load_inline(
+            name="mylib",
```

Comment:
I think you're triggering the versioning check. (https://github.com/pytorch/pytorch/blob/main/torch/utils/cpp_extension.py#L1801).

What you could do is take first_graph_break and sanitize it (e.g. replace mylib_v4312134 with mylib) and then do an expecttest with it